src/main/java/picard/illumina/CheckIlluminaDirectory.java

```diff
@@ -161,42 +165,44 @@ protected int doWork() {
 
                 //check s.locs
                 final File locsFile = new File(BASECALLS_DIR.getParentFile(), AbstractIlluminaPositionFileReader.S_LOCS_FILE);
-                final LocsFileReader locsFileReader = new LocsFileReader(locsFile);
-                final List<AbstractIlluminaPositionFileReader.PositionInfo> locs = new ArrayList<>();
-                while (locsFileReader.hasNext()) {
-                    locs.add(locsFileReader.next());
+                final List<AbstractIlluminaPositionFileReader.PositionInfo> locs;
+                final Map<Integer, File> filterFileMap;
+                try (LocsFileReader locsFileReader = new LocsFileReader(locsFile)) {
+                    locs = new ArrayList<>();
```

Comment:
Woah... this works even when it's `final`?

Why not just assign the `new ArrayList` at the point where you declare the var?

Reply:
its declared final and then assigned. It just can't be re-assigned.